Softeq.XToolkit.WhiteLabel.Forms/Controls/ToggleButton.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using System.Runtime.CompilerServices;
+using Xamarin.Forms;
+
+namespace Softeq.XToolkit.WhiteLabel.Forms.Controls
+{
+    /// <summary>
+    ///     A button with On/Off states. State names for VisualStateManager are
+    ///     <see cref="ToggledOnState"/>, <see cref="ToggledOffState"/>.
+    /// </summary>
+    public class ToggleButton : Button
```

Comment:
It seems like `Behavior`. Have you tried this?

Behavior can be used for `Button`, `ImageButton` ...

Reply:
I will look into this. The idea was that you can provide any styles for On/Off states so it can actually be a button with only ImageSource that changes (kinda the same as ImageButton) - I was actually using it as a custom check box that way. I was also initially using it in a custom tab selection control (with only 1 selected at a time) where these buttons were created programmatically depending on the amount of tabs needed and that was controlling their states - I guess that's why it was a better idea to make it a button extension and not behavior. I will think of turning it into behavior.

Reply:
@wcoder the problem with Behavior is that Button and ImageButton only have common accessible ancestor View which does not have Clicked event (ImageButton does not inherit from Button as I would expect and IButtonElement is not accessible) so I do not see how I can make this behavior shared for Button and ImageButton. I still can make it a behavior for Button but I'm not sure if this makes sense

Reply:
Agree